### ISSUE TITLE
Password field: "reveal/hide password" is not keyboard accessible

### ISSUE THREAD

**Issue by d-m-s** [vmware/clarity#3122]
## Describe the bug
Password field: "reveal/hide password" is not keyboard accessible
(the icon cannot be navigated with the keyboard)

## How to reproduce
1.  Go to https://clarity.design/documentation/password
2.  Click in the basic password example
3.  Press TAB-key

Result: Focus navigates to the next password field.

## Expected behavior
Focus should move to the eye-icon
`<space>, <enter>` key should reveal/hide password.
-> Similar to Datepicker https://clarity.design/documentation/datepicker

## Versions
* Clarity: 1.0
* Device: Desktop, Notebook
* OS: Windows 10
* Browser: Chrome
